blocks/library/gallery/gallery-image.js

```diff
@@ -147,6 +148,6 @@ class GalleryImage extends Component {
 	}
 }
 
-export default withAPIData( ( { id } ) => ( {
-	image: id ? `/wp/v2/media/${ id }` : {},
+export default withSelect( ( select, { id } ) => ( {
```

Comment:
Minor: This works well for an in-place replacement, but in my own experience I want to start going out of my way to be more verbose with how we use `withSelect`, to set a good precedent especially when scaling a component to select many things from the same store ([example](https://github.com/WordPress/gutenberg/blob/f4cc3bfb4fd751dbd7e711868088ef33f710ff11/edit-post/components/layout/index.js#L95-L103)), and to avoid both (a) in-place `ownProps` destructuring (destructuring obscures _what_ is being destructured when used in arguments) and (b) the arrow function implicit return of objects (difficult to discern between implicit return and a function body).

This is how I would write:

```js
export default withSelect( ( select, ownProps ) => {
	const { getMedia } = select( 'core' );
	const { id } = ownProps;

	return {
		image: id ? getMedia( id ) : null,
	};
} )( GalleryImage );
```

Or:

```js
export default withSelect( ( select, ownProps ) => {
	const { getMedia } = select( 'core' );
	const { id } = ownProps;
	if ( ! id ) {
		return;
	}

	return {
		image: getMedia( id ),
	};
} )( GalleryImage );
```